static/app/utils/theme/theme.chonk.tsx

```diff
@@ -226,13 +161,33 @@ type Next<R extends ValidLengthArgument> = NextMap[R];
  * return a single color, not two colors. It smells like either a bug or off by one error.
  * @param length - The number of series to return a color palette for?
  */
-function getChartColorPalette<Length extends ValidLengthArgument>(
+function makeChartColorPalette<T extends ChartColorPalette>(
+  palette: T
+): <Length extends ValidLengthArgument>(
   length: Length | number
-): Exclude<ChartColorPalette[Next<Length>], undefined> {
-  // @TODO(jonasbadalic) we guarantee type safety and sort of guarantee runtime safety by clamping and
-  // the palette is not sparse, but we should probably add a runtime check here as well.
-  const index = Math.max(0, Math.min(CHART_PALETTE.length - 1, length + 1));
-  return CHART_PALETTE[index] as Exclude<ChartColorPalette[Next<Length>], undefined>;
+) => Exclude<ChartColorPalette[Next<Length>], undefined> {
+  return function getChartColorPalette<Length extends ValidLengthArgument>(
+    length: Length | number
+  ): Exclude<ChartColorPalette[Next<Length>], undefined> {
+    // The new palette has less elements than the old one, so we will loop colors to fill the palette space.
```

Comment:
If looping colors is what we want, wouldn’t it be better to just statically add the colors we’re looping to the palette so that they get the same length again? This would allow us to keep the type-safety I think.

Then, once chonk is the only available theme, we can shorten the palette and we would get errors on all invocations and refactor them to pick the correct colors.

Reply:
We wont be able to shorten the palette unfortunately because we have dynamic palette in some places that depends on the data we receive from the server, so we'll always need either a clamping or loop mechanism to generate the colors for all series. This might be a good place to add some monitoring though, because we may see that we never exceed a certain palette size